packages/near-snap/src/lib/near-snap.ts

```diff
@@ -0,0 +1,113 @@
+  InjectedWallet,
+} from "@near-wallet-selector/core";
+import { isMobile } from "is-mobile";
+import { enable, getSnapAccounts, signTransactions } from "./methods";
+import { isMetaMaskAvailable } from "./utils";
+import icon from "../lib/icon";
+
+const isDev = false;
+
+const NearSnapWallet: WalletBehaviourFactory<InjectedWallet> = async ({
+  options,
+  provider,
+  store,
+}) => {
+  const network = options.network.networkId as "testnet" | "mainnet";
+
+  return {
+    async signIn() {
+      // Sign in to My Wallet for access to account(s).
+      await enable(isDev);
+
+      return await getSnapAccounts(isDev, network);
+    },
+
+    async signOut() {
+      // Nothing to do here!!
+    },
+
+    async getAccounts() {
+      return await getSnapAccounts(isDev, network);
+    },
+
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    async verifyOwner() {},
+
+    async signAndSendTransaction({ receiverId, actions }) {
+      const accountPayload = await getSnapAccounts(isDev, network);
+      const { accountId, publicKey } = accountPayload[0];
+
+      const accessKey = await provider.viewAccessKey({ accountId, publicKey });
+
+      const { contract } = store.getState();
+      let nonce = parseInt(accessKey.nonce.toString());
+      if (!contract) {
+        throw new Error("Wallet not signed in");
+      }
+
+      const signedTxs = await signTransactions(isDev, network, [
+        {
+          receiverId: receiverId || contract.contractId,
+          actions,
+          nonce: ++nonce,
+          recentBlockHash: accessKey.block_hash,
+        },
+      ]);
+
+      return await provider.sendTransaction(signedTxs[0]);
+    },
+
+    async signAndSendTransactions({ transactions }) {
+      const accountPayload = await getSnapAccounts(isDev, network);
+      const { accountId, publicKey } = accountPayload[0];
+      const accessKey = await provider.viewAccessKey({ accountId, publicKey });
+      let nonce = parseInt(accessKey.nonce.toString());
+
+      const signedTxs = await signTransactions(
+        isDev,
+        network,
+        transactions.map(({ receiverId, actions }) => ({
+          receiverId,
+          actions,
+          nonce: ++nonce,
+          recentBlockHash: accessKey.block_hash,
+        }))
+      );
+
+      return Promise.all(signedTxs.map((tx) => provider.sendTransaction(tx)));
+    },
+  };
+};
+
+export interface NearSnapParams {
+  iconUrl?: string;
+  deprecated?: boolean;
+}
+
+export const setupNearSnap = ({
+  iconUrl = icon,
+  deprecated = false,
+}: NearSnapParams = {}): WalletModuleFactory<InjectedWallet> => {
+  return async () => {
+    if (isMobile()) {
+      return null;
+    }
+
+    return {
+      id: "near-snap",
+      type: "injected",
+      metadata: {
+        name: "NearSnap",
+        description: null,
+        iconUrl,
+        deprecated,
+        downloadUrl: "https://metamask.io/flask/",
+        available: await isMetaMaskAvailable(),
```

Comment:
The `await isMetaMaskAvailable()` is blocking the execution.

Could you please fix this?

Reply:
Hi,
I can't reproduce the issue of blocked execution, can you share more on what's causing the issue.

Do you have metamask flask installed, or regular metamask? You should have metamask flask for snap to work.

`isMetaMaskAvailable()` checks if metamask is installed, and calls metamask flask exclusive method 
(
```
await window.ethereum.request({
    method: "wallet_getSnaps",
  });
```
) 
to check if installed metamask is flask version.
Either way if it fails it just returns false if there is no metamask flask installed, and should not block rest of the app.

Reply:
Hey, I know Metamask and Metamask Flask should not be active at the same time.

I had this "Near Snap" added to the Metamask Flask before back then when I did the review for the PR but yesterday I could not even try this option because it's blocking the execution so here's what I did to start "fresh".

- Remove Metamask
- Remove Metamast Flask
- Install again Metamask Flask

Here's what's happening:

![blocked](https://user-images.githubusercontent.com/95851345/217215991-5c179ad6-e3e4-4091-b05c-7303f7f72157.png)

After some more investigation somewhere down the line the `await isMetaMaskAvailable()` is calling a function `await getWalletSnaps()` which is just hanging there and never resolving.

I thought this could be an issue just for me but other team members tried it and they had the same experience.

Reply:
For me loader spins for 5 seconds and then shows Log in. Snap works as intended afterwards.

What are the console errors?
Is the metamask flask version `10.23.2-flask.0`?

Reply:
Hey, @irubido I have this version of Metamask Flask `10.23.2-flask.0` but I just found out what the issue is.

Somehow the having [MetaMask Flask](https://chrome.google.com/webstore/detail/metamask-flask-developmen/ljfoeinjpaedjfecbmggjgodbgkmjkjk) and [Coin98](https://chrome.google.com/webstore/detail/coin98-wallet/aeachknmefphepccionboohckonoeemg) extensions installed at the same time is blocking the execution.

I disabled the `Coin98` just for now and tested the `Near Snap` now everything is working I could:
`signIn`, `signOut`, `signAndSendTransaction(s)` with and without deposit.

Coin98 has been added to the wallet selector a few months ago.

Could you please investigate what's the conflict between `Coin98` and `Near Snap`?

Reply:
Interesting, I'll take a look into it

Reply:
coin98 overrides `winsow.etherium` so the request is going to coin98 wallet, not to meta mask, and coin98 does not throw an error on not existing RPC call, but just hang out forever

Reply:
coin98 overrides `winsow.ethereum` so the request is going to coin98 wallet, not to meta mask, and coin98 does not throw an error on not existing RPC call, but just hangs out forever

Reply:
did some research with Coin98 wallet and they actually change methods and leave object unchanged
![image](https://user-images.githubusercontent.com/6376047/217347503-32238e8b-9562-4d67-b839-db111113b0d3.png)
the issue is that there is a variable `isMetaMask` but the app is actually communicating with Coin98 wallet

Did some "hacking" to resolve this issue. (for curios one added promise race timeout 1 sec)
Think that now is testable.

cc @kujtimprenkuSQA

Reply:
@BeroBurny thank you for taking the time to add the fix. Will test it out first thing in the morning.

Reply:
Sorry for the late response, this looks much better at least the execution is not being blocked.

Even this way `Near Snap` is living somehow under the shadow of `Coin98`, if **Coin98** is installed this means the **Near Snap** will not work and will show the `WalletNotInstalled` view inaccurately.

We have informed the Coin98 team and we hope they prioritize this issue.

In the meantime can you resolve these 2 comments:
https://github.com/near/wallet-selector/pull/460#discussion_r1048523512
https://github.com/near/wallet-selector/pull/460#discussion_r1048536752

Reply:
Step into contact with the MetaMask team to find a solution to that problem.
Meantime did some research additional to find an acceptable solution for a problem, but non without unreadable magic (like editing static `*.html`)

Reply:
Step into contact with the MetaMask team to find a solution to that problem.
Meantime did some research additional to find an acceptable solution for a problem, but non without unreadable magic (like editing static `*.html`)

Reply:
@BeroBurny & @irubido thank you again for submitting this PR and resolving all the comments.

We have decided to merge this PR but wanted to let you know the response we got from the `Coin98` team.

Coin98 overrides `window.ethereum` to facilitate their user's connection to dApps. The current solution for this is to turn `off` the override, following either of these ways:

1. Through the "Override Wallet" feature on Coin98 Extension (Home > Settings > Override Wallet). Turning off the button means allowing the other wallet to override Coin98 and vice versa.
![index](https://user-images.githubusercontent.com/95851345/221811561-9fac7371-624c-4a38-ad35-223aab60e281.jpg)

2. Through Chrome's "Manage Extensions" section > Deactivate Coin98 when wishing to use other wallets.

We will document this after merging this PR.